src/compiler/scala/reflect/macros/compiler/Validators.scala

```diff
@@ -51,23 +51,23 @@ trait Validators {
       // we only check strict correspondence between value parameterss
       // type parameters of macro defs and macro impls don't have to coincide with each other
       if (aparamss.length != rparamss.length) MacroImplParamssMismatchError()
-      map2(aparamss, rparamss)((aparams, rparams) => {
+      foreach2(aparamss, rparamss)((aparams, rparams) => {
         if (aparams.length < rparams.length) MacroImplMissingParamsError(aparams, rparams)
         if (rparams.length < aparams.length) MacroImplExtraParamsError(aparams, rparams)
       })
 
       try {
         // cannot fuse this map2 and the map2 above because if aparamss.flatten != rparamss.flatten
         // then `atpeToRtpe` is going to fail with an unsound substitution
-        map2(aparamss.flatten, rparamss.flatten)((aparam, rparam) => {
+        foreach2(aparamss, rparamss){ foreach2(_, _){ (aparam, rparam) => {
```

Comment:
I thought about it at first, but having it would transform the code above into: 
```Scala
foreach2(aparamss, rparamss){ foreach2(_, _){ (aparam, rparam) => {
mforeach2(aparamss, rparamss){ (aparam, param) =>
```

Not sure if removing the 15 characters is worth it. On the other hand, I do not know if avoiding the inner lambda would be worth it.